File: backend/app/api/routes/cash_flow.py
from datetime import date as date_

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_session
from app.schemas.cash_flow import CashFlowResponse
from app.services.cash_flow_service import get_cash_flow

router = APIRouter(prefix='/cash-flow', tags=['cash-flow'])


@router.get('', response_model=CashFlowResponse)
async def read_cash_flow(
  start_date: date_ | None = Query(default=None),
  end_date: date_ | None = Query(default=None),
  session: AsyncSession = Depends(get_session),
) -> CashFlowResponse:
  return await get_cash_flow(session, start_date, end_date)

File: frontend/src/api/cashFlow.ts
import { api } from '@/api/client';
import type { CashFlowResponse } from '@/types';

export function fetchCashFlow(startDate?: string, endDate?: string) {
  const params = new URLSearchParams();
  if (startDate) params.set('start_date', startDate);
  if (endDate) params.set('end_date', endDate);
  const query = params.toString();
  return api.get<CashFlowResponse>(`/cash-flow${query ? `?${query}` : ''}`);
}
